### DISCUSSION TITLE
ERROR: cannot execute SELECT FOR UPDATE in a read-only transaction

### DISCUSSION THREAD

**Discussion by boskiv** [laravel/framework#38164] [Q&A]
I noticed that in my production instance 3 weird events come to Sentry that night.
```
Illuminate\Database\QueryException
SQLSTATE[25006]: Read only sql transaction: 7 ERROR:  cannot execute SELECT FOR UPDATE in a read-only transaction (SQL: select * from "jobs" where "queue" = default and (("reserved_at" is null and "available_at" <= 1627474572) or ("reserved_at" <= 1627474482)) order by "id" asc limit 1 FOR UPDATE SKIP LOCKED)
```
Trace:
```
Crashed in non-app: /vendor/laravel/framework/src/Illuminate/Database/Connection.php in Illuminate\Database\Connection::runQueryCallback
/artisan at line 37

$kernel = $app->make(Illuminate\Contracts\Console\Kernel::class);
$status = $kernel->handle(
    $input = new Symfony\Component\Console\Input\ArgvInput,
    new Symfony\Component\Console\Output\ConsoleOutput
);
/*
|--------------------------------------------------------------------------
| Shutdown The Application
```

Command:
```
Starting Artisan command: queue: work
{
input: 'queue:work' --once --name=default --queue=default --backoff=0 --memory=128 --sleep=3 --tries=1
}
```

Is it ok? Should I start to care about it?

**Comment by devplaylogiq** [comment]
same happening here after set up read and write hosts at database connection configuration at config/database.php
laravel 8.83.27

all my scheduled jobs re queued and processed successfully but i see this on the log error file

**Reply by PEscar** [reply]
does anyone know how/when laravel decide if a query should be executed on the read or the write host ?
because this query should be executed on the write host, not on the read one:
select * from `jobs` where `queue` = default and ((`reserved_at` is null and `available_at` <= 1676581200) or (`reserved_at` <= 1676581110)) order by `id` asc limit 1 FOR UPDATE SKIP LOCKED
